x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/mappings_validator.test.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { validateMappings, validateProperties, isObject } from './mappings_validator';
+
+jest.mock('ui/index_patterns', () => ({
+  ILLEGAL_CHARACTERS: '',
+  validateIndexPattern: () => ({}),
+}));
+
+describe('Mappings configuration validator', () => {
+  it('should convert non object to empty object', () => {
```

Comment:
I'm a little confused about this behavior. Are these input values being _converted_ to something or just replaced with empty objects? In either case I'm not sure why we want this behavior. Would you mind explaining?

Reply:
We are making sure that we are dealing with objects in the first place. If we are not then there is no need to go any further in the validation (and the rest of the validation can safely call object-specific method like `hasOwnProperty`). It is a sanitization.